src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -1655,3 +1665,246 @@ def cte_statement(queries: List[str], table_names: List[str]) -> str:
+            chunk_path = os.path.join(tmp_folder, f"{table_name}_{file_number}.parquet")
+            pyarrow.parquet.write_table(
+                table.slice(offset=offset, length=chunk_size),
+                chunk_path,
+                **kwargs,
+            )
+            # upload chunk
+            upload_sql = (
+                "PUT /* Python:snowflake.snowpark._internal.analyzer.analyzer_utils.write_arrow() */ "
+                "'file://{path}' @{stage_location} PARALLEL={parallel}"
+            ).format(
+                path=chunk_path.replace("\\", "\\\\").replace("'", "\\'"),
+                stage_location=stage_location,
+                parallel=parallel,
+            )
+            cursor.execute(upload_sql, _is_internal=True)
+            # Remove chunk file
+            os.remove(chunk_path)
+
+    if quote_identifiers:
+        quote = '"'
+        snowflake_column_names = [str(c).replace('"', '""') for c in table.schema.names]
+    else:
+        quote = ""
+        snowflake_column_names = list(table.schema.names)
+    columns = quote + f"{quote},{quote}".join(snowflake_column_names) + quote
+
+    def drop_object(name: str, object_type: str) -> None:
+        drop_sql = f"DROP {object_type.upper()} IF EXISTS {name} /* Python:snowflake.snowpark._internal.analyzer.analyzer_utils.write_arrow() */"
+        cursor.execute(drop_sql, _is_internal=True)
+
+    if auto_create_table or overwrite:
+        file_format_location = _create_temp_file_format(
+            cursor,
+            database,
+            schema,
+            quote_identifiers,
+            compression_map[compression],
+            sql_use_logical_type,
+            use_scoped_temp_object,
+        )
+        infer_schema_sql = f"SELECT COLUMN_NAME, TYPE FROM table(infer_schema(location=>'@{stage_location}', file_format=>'{file_format_location}'))"
+        infer_result = cursor.execute(infer_schema_sql, _is_internal=True)
+        assert infer_result is not None
+        column_type_mapping = dict(infer_result.fetchall())  # pyright: ignore
+        create_table_columns = ", ".join(
+            [
+                f"{quote}{snowflake_col}{quote} {column_type_mapping[col]}"
+                for snowflake_col, col in zip(
+                    snowflake_column_names, table.schema.names
+                )
+            ]
+        )
+
+        target_table_location = build_location_helper(
+            database,
+            schema,
+            random_name_for_temp_object(TempObjectType.TABLE)
+            if (overwrite and auto_create_table)
+            else table_name,
+            quote_identifiers,
+        )
+
+        create_table_sql = (
+            f"CREATE {table_type.upper()} TABLE IF NOT EXISTS {target_table_location} "
+            f"({create_table_columns})"
+            f" /* Python:snowflake.snowpark._internal.analyzer.analyzer_utils.write_arrow() */ "
+        )
+        cursor.execute(create_table_sql, _is_internal=True)
```

Comment:
Since this is a new implementation, we should create table when `auto_create_table` is true. This change is scheduled as a BCR for python connector but there is no major development happening there.